internal/plugin/factory.go

```diff
@@ -95,6 +96,65 @@ func BuiltinFactory(name string, typ component.Type) interface{} {
 	return Factory(cmd, typ)
 }
 
+// UnmanagedPluginFactory produces a provider factory that uses the passed
+// reattach information to connect to go-plugin processes that are already
+// running, and implements Instance against it.
+func UnmanagedPluginFactory(reattach *plugin.ReattachConfig, typ component.Type) interface{} {
```

Comment:
`unmanaged` is a bit confusing since go-plugin actually does support a concept of managed vs. unmanaged plugins (defined as whether it auto cleans them up).

My recommendation would be changing this to `ReattachPluginFactory` cause its reattaching.